### ISSUE TITLE
Comm Panel deals with same email addresses stupidly

### ISSUE THREAD

**Issue by wnavarre** [learning-unlimited/ESP-Website#1849]
If I select one of my student test accounts to receive the message (specifying by USERNAME), it will say that there are 3 unique users going to receive the email (since there are three student accounts with that email?). 

Sure enough, after I send, I get three emails in my inbox: one for each of my student accounts.

**Comment by benjaminjkraft** [comment]
This is a bit trickier than just deduplicating, because if the multiple accounts represent multiple students all using their parent's email address, and the email is e.g. financial aid, we do want to make sure to send a copy for each account.  Probably an option to do it either way would be the best approach.

**Comment by benjaminjkraft** [comment]
Oh, sorry, I totally misunderstod.  Ignore the above comment.

**Comment by wnavarre** [comment]
Right. I entered a **USERNAME** to send to, not an **EMAIL**.

If I had entered my email navarre@mit.edu, then the behaviour would be
exactly correct.

~~William Navarre

**Comment by benjaminjkraft** [comment]
Just to double-check, the username was not a substring of the other accounts' usernames?  We probably do a `__contains` rather than a `__exact` lookup.

**Comment by wnavarre** [comment]
Oh it probably was.

I would call that a bug. I can imagine use cases where you want to email a
particular student something. I cannot imagine use cases where you'd want
to email every student whose username contains something given that we
don't assign usernames and they aren't meaningful except that as a *whole*
they uniquely identify a student. .

Do such use cases exist?

~~William Navarre

**Comment by benjaminjkraft** [comment]
Yeah, seems reasonable.  I'll edit the issue description for future clarity.

**Comment by willgearty** [comment]
Actually, Stanford sometimes creates accounts for outreach students with set prefixes which are useful for filtering. I'm not sure we ever actually email those students through the comm panel, but the possibility exists.

I know that we can use "|" to include multiple usernames, which seems like a regular expression thing, so why can't we just parse other regular expressions, like "*" or "."?

**Comment by benjaminjkraft** [comment]
We probably could!

**Comment by yashikan-dev** [comment]
Hi @willgearty , I would like to work on this issue

**Comment by willgearty** [comment]
Unassigning due to inactivity.
